engine/src/main/java/org/camunda/bpm/engine/impl/bpmn/behavior/ParallelGatewayActivityBehavior.java

```diff
@@ -63,18 +66,52 @@ public void execute(ActivityExecution execution) throws Exception {
     lockConcurrentRoot(execution);
     
     List<ActivityExecution> joinedExecutions = execution.findInactiveConcurrentExecutions(activity);
-    int nbrOfExecutionsToJoin = execution.getActivity().getIncomingTransitions().size();
-    int nbrOfExecutionsJoined = joinedExecutions.size();
+    
+    List<ActivityExecution> finalExecutions = new ArrayList<ActivityExecution>();
+    
+    Map<String, List<ActivityExecution>> mappedExecutions = mapExecutionsToTransitions(joinedExecutions, finalExecutions);
+    
+    List<PvmTransition> incomingTransitions = execution.getActivity().getIncomingTransitions();
+    for (PvmTransition incomingTransition : incomingTransitions) {
+      List<ActivityExecution> sequenceExecutions = mappedExecutions.get(incomingTransition.getId());
+      if (sequenceExecutions == null || sequenceExecutions.isEmpty()){
+        //TODO if required for logging we can go through all of them?
+        break;
```

Comment:
You are right the break will not work with the case that you are saying, I think that even with 2 incoming sequence flows it will be the same. If one execution has reached the upgrade before the upgrade the transitionId will be null and it will always break the for. 

I will make a test case to confirm this and I will remove the break and put another check with the finalExecutions size.